lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -3605,9 +3605,16 @@ public function mapEmbedded(array $mapping)
     {
         $this->assertFieldNotMapped($mapping['fieldName']);
 
+        if (! isset($mapping['class']) && $this->isTypedProperty($mapping['fieldName'])) {
+            $type = $this->reflClass->getProperty($mapping['fieldName'])->getType();
+            if ($type instanceof ReflectionNamedType) {
+                $mapping['class'] = $type->getName();
+            }
+        }
+
         $this->embeddedClasses[$mapping['fieldName']] = [
             'class' => $this->fullyQualifiedClassName($mapping['class']),
-            'columnPrefix' => $mapping['columnPrefix'],
+            'columnPrefix' => $mapping['columnPrefix'] ?? null,
```

Comment:
:thinking: what does this have to do with the rest of your PR?

Reply:
I assume phpstan failure they mentioned in the PR description.